src/cargo/core/source/source_id.rs

```diff
@@ -143,7 +153,7 @@ impl SourceId {
             }
             "sparse" => {
                 let url = string.into_url()?;
-                Ok(SourceId::new(SourceKind::Registry, url, None)?
+                Ok(SourceId::new(SourceKind::SparseRegistry, url, None)?
```

Comment:
I'm wondering if it might help to not keep the `sparse+` in the url. Keeping it there makes `SourceKind::SparseRegistry` different from all the other kinds.

From skimming over, it seems like to make that more consistent, `remote_source_kind` will need to strip the prefix, and return both the kind and url. But otherwise it looks like that should have relatively minor fallout. Do you think that might work?

Reply:
Yes, I like it.